### PR TITLE
fixed encoding bug. thanks to franky and cccamp15.

### PR DESCRIPTION
**Pull request by einsiedlerkrebs**
it is not my work, but i heard from others about the bug and wanted to commit it.

### CODE REVIEW

**Comment by jcw** [comment]
Thanks, I'll commit it - though I don't really understand what the difference is - incorrect return value? The "= {}" can't make any difference, I think.

**Comment by einsiedlerkrebs** [comment]
as i far as i remember, the issue is that it does not encode correctly
and it always produce different output. The guy who fixed it, said that
it might be depending on the compliler, if it works or not.

File: EmBencode.h [hunk 1]
```diff
@@ -62,11 +62,19 @@ class EmBencode {
...
-    PushData(ultoa(num, buf, 10), strlen(buf));
+    char buf[11] = {};
+    ultoa(num, buf, 10);
+    PushData(buf, strlen(buf));
   }
 
+  //doesn't work always
+//  static void PushCount (uint32_t num) {
+//    char buf[11];
+//    PushData(ultoa(num, buf, 10), strlen(buf));
```
Comment (einsiedlerkrebs) [inline EmBencode.h]: you can not make sure if ultoa or strlen is used first.

**Comment by jcw** [comment]
Aha! Yes, that explains it all - thanks again.

**Comment by einsiedlerkrebs** [comment]
re, for your work!

**Pull request merged**